public/app/features/query/state/queryAnalytics.ts

```diff
@@ -28,6 +28,7 @@ export function emitDataRequestEvent(datasource: DataSourceApi) {
       datasourceUid: datasource.uid,
       datasourceType: datasource.type,
       dataSize: 0,
+      panelId: 0,
```

Comment:
This will send panelId as `0` by default if it's not a number, not sure if this is an expected behavior?

Reply:
0 is the nil and default value in the backend for this DTO, so I guess it's expected

Reply:
dataSize seems to follow the same approach